src/MSALWrapper.Test/AuthFlow/AuthFlowExecutorTest.cs

```diff
@@ -751,7 +820,36 @@ private EquivalencyAssertionOptions<AuthFlowResult> ExcludeDurationTimeSpan(Equi
         private AuthFlowExecutor Subject(IEnumerable<IAuthFlow> authFlows)
         {
             var logger = this.serviceProvider.GetService<ILogger<AuthFlowExecutor>>();
-            return new AuthFlowExecutor(logger, authFlows);
+            return new AuthFlowExecutor(logger, authFlows, this.stopwatch);
+        }
+
+        // This auth flow is for testing timeouts. It should always timeout.
+        private class AlwaysTimesOutAuthFlow : IAuthFlow
+        {
+            public async Task<AuthFlowResult> GetTokenAsync()
+            {
+                await Task.Delay(TimeSpan.FromMinutes(100));
+                return null;
+            }
+        }
+
+        // This auth flow is used to wait before returning error.

```

Comment:
This class we don't need to add since we can easily mock this behavior directly in the test. because we are not relying on this to pass any time, we have mocked the stop watch - the mock behavior can immediately return the error.